bundles/org.openhab.binding.carnet/src/main/java/org/openhab/binding/carnet/internal/handler/CarNetAccountHandler.java

```diff
@@ -0,0 +1,295 @@
+     *
+     * @param listener Listener interface provided by Vehicle Handler
+     */
+    public void registerListener(CarNetDeviceListener listener) {
+        vehicleInformationListeners.add(listener);
+    }
+
+    /**
+     * Called by vehicle handler to unregister callback
+     *
+     * @param listener Listener interface provided by Vehicle Handler
+     */
+    public void unregisterListener(CarNetDeviceListener listener) {
+        vehicleInformationListeners.remove(listener);
+    }
+
+    /**
+     * Forward discovery information to all listeners (Vehicle Handlers)
+     *
+     * @param vehicleInformationList
+     */
+    private void informVehicleInformationListeners(@Nullable List<CarNetVehicleInformation> vehicleInformationList) {
+        this.vehicleInformationListeners.forEach(discovery -> discovery.informationUpdate(vehicleInformationList));
+    }
+
+    /**
+     * Notify all listeners about status changes
+     *
+     * @param status New status
+     * @param detail Status details
+     * @param message Message
+     */
+    void stateChanged(ThingStatus status, ThingStatusDetail detail, String message) {
+        updateStatus(status, detail, message);
+        this.vehicleInformationListeners.forEach(discovery -> discovery.stateChanged(status, detail, message));
+    }
+
+    /**
+     * Thing configuration was updated
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+    }
+
+    /**
+     * Empty handleCommand for Account Thing
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getIdWithoutGroup();
+        logger.debug("Handle command '{}' for channel {}", command, channelId);
+        if (command == RefreshType.REFRESH) {
+            return;
+        }
+    }
+
+    /**
+     * Sets up a polling job (using the scheduler) with the given interval.
+     *
+     * @param initialWaitTime The delay before the first refresh. Maybe 0 to immediately
+     *            initiate a refresh.
+     */
+    private void setupRefreshJob(int initialWaitTime) {
+        cancelRefreshJob();
+        logger.trace("Setting up token refresh job, checking every 5 minutes");
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> refreshToken(), initialWaitTime,
+                API_TOKEN_REFRESH_INTERVAL_SEC, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Cancels the polling job (if one was setup).
+     */
+    private void cancelRefreshJob() {
+        if (refreshJob != null) {
+            refreshJob.cancel(false);
+        }
+    }
+
+    private void refreshToken() {
+        logger.debug("Validating/refreshing tokens");
+        try {
+            api.refreshTokens();
+        } catch (CarNetException e) {
+            logger.debug("Unable to refresh tokens", e);
```

Comment:
Logging stack trace.

Reply:
that's the intention

Reply:
It seems a bit unreasonable to log the stack trace when the network fails.